airflow/providers/amazon/aws/operators/sagemaker_base.py

```diff
@@ -97,3 +97,8 @@ def preprocess_config(self):
 
     def execute(self, context):
         raise NotImplementedError('Please implement execute() in sub class!')
+
+    @cached_property
+    def hook(self):
```

Comment:
Well, I liked this one :)

Reply:
Me,too. I would like to rewrite all get_hook() to this way but I think it's out of scope this work, so I keep the original form.